lib/runtime-c-api/src/lib.rs

```diff
@@ -208,14 +208,15 @@ pub extern "C" fn wasmer_memory_grow(
     delta: uint32_t,
 ) -> wasmer_result_t {
     let memory = unsafe { &*(memory as *mut Memory) };
-    let maybe_delta = memory.grow(Pages(delta));
-    if let Some(_delta) = maybe_delta {
-        wasmer_result_t::WASMER_OK
-    } else {
-        update_last_error(CApiError {
-            msg: "unable to grow memory".to_string(),
-        });
-        wasmer_result_t::WASMER_ERROR
+    let delta_result = memory.grow(Pages(delta));
+    match delta_result {
+        Ok(_) => wasmer_result_t::WASMER_OK,
+        Err(_) => {
+            update_last_error(CApiError {
+                msg: "unable to grow memory".to_string(),
+            });
```

Comment:
If the error implements "Error",  you can remove this CApiError and just pass the `Err(error)` to `update_last_error(error)`

Reply:
Right! Done. ✔️